sstables/mx/bsearch_clustered_cursor.hh

```diff
@@ -461,6 +461,22 @@ public:
     { }
 
     future<std::optional<skip_info>> advance_to(position_in_partition_view pos) override {
+        return do_advance_to(pos, false);
+    }
+
+    future<> warm_up(position_in_partition_view pos) override {
+        auto old_current_pos = _current_pos;
+        auto old_current_idx = _current_idx;
+        auto old_upper_idx = _upper_idx;
+        return do_advance_to(pos, true).then([this, old_current_idx, old_upper_idx, old_current_pos] (auto) {
+            _current_pos = std::move(old_current_pos);
+            _current_idx = old_current_idx;
+            _upper_idx = old_upper_idx;
+        });
+    }
+
+private:
+    future<std::optional<skip_info>> do_advance_to(position_in_partition_view pos, bool dry_run = false) {
```

Comment:
Isn't a dry run wasteful in CPU? Could be worthwhile, but I don't understand what it accomplishes.

Reply:
Yes. For the cases where it helps, it's worthwhile. My tests with 1M rows show it's worthwhile, we save more cycles on avoiding the IO. But it depends.